api/src/main/java/com/datastrato/gravitino/exceptions/UserAlreadyExistsException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.exceptions;
+
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+/** An exception thrown when a resource already exists. */
+public class UserAlreadyExistsException extends AlreadyExistsException {
```

Comment:
User may have same name, how do you support it?

Reply:
User must have different name. It's an identifier for the users.

Reply:
I don't accept the reply. I think you should give an explanation on how to support this case.

Reply:
> I don't accept the reply. I think you should give me an explain of how to support this case.

OAuth server will use the name of user as the identifier. If we add too many concepts, it may be strange for users. Referring to Snowflake, we can add the concept `display name`.

Reply:
Then you should add it as a field. You should carefully think about your user model and refer to some other projects to add some necessary fields to the entity. From my current feeling, I think the current `user` model seems too simple.

Reply:
How do you support different login systems, like email, sso, oauth? They may have different requirements for user model, you should carefully think of.

Reply:
From my considerition,
For Saas service, we should use email as the name of user. This is the way which Databricks used.
For precise deployment, almost all companies will give every staff a unique name.